Source/Request.swift

```diff
@@ -434,14 +434,19 @@ public class Request {
     /// Called to determine whether retry will be triggered for the particular error, or whether the instance should
     /// call `finish()`.
     ///
-    /// - Parameter error: The possible `Error` which may trigger retry.
-    func retryOrFinish(error: Error?) {
+    /// - Parameter error: The possible `AFError` which may trigger retry.
+    func retryOrFinish(error: AFError?) {
         guard let error = error, let delegate = delegate else { finish(); return }
 
         delegate.retryResult(for: self, dueTo: error) { retryResult in
             switch retryResult {
-            case .doNotRetry, .doNotRetryWithError:
-                self.finish(error: retryResult.error)
+            case .doNotRetry:
+                self.finish()
+            case .doNotRetryWithError(let retryError):
+                guard let retryError = retryError.asAFError else {
```

Comment:
Since this is actually called a few places, perhaps an additional convenience property is in order:
```swift
var mustBeAFError: AFError {
    guard let afError = self as? AFError else {
        fatalError("Error must be AFError")
    }

    return afError
}
```
I'm not entirely certain of the name, but something like that.

Reply:
What about `asAFErrorOrFail`?

Reply:
Sounds good.

Reply:
Actually, a more useful one would be:
```swift
func asAFError(orFailWith message: @autoclosure () -> String) {
    guard let afError = self as? AFError else {
        fatalError(message())
    }
    return afError
}